### ISSUE TITLE
How should keyboard focus be handled for Tabs with aria-disabled?

### ISSUE THREAD

**Issue by jchiem** [w3c/aria-practices#3370]
This question is related to #2318 and the recommendations on [Focusability of disabled controls](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols) and [Example of Automatic tabs](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/ ) or [Example of Manual tabs](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-manual/) 

The APG pattern suggests that tabs should be reachable via the tab key, and then navigated using arrow keys. If a tab is disabled using aria-disabled, how should that behave in terms of keyboard navigation? Isn't it conflicting to make tabs focusable but have the tab pattern show tabs as not keyboard navigatable with the "tab key" ?

The current example shows that tabs should focus the first focusable tab within the tabs if you use `TAB` on your keyboard to navigate, and then subsequent `TAB` key presses should focus either the associated tab panel (or if there are focusable elements within the active tab panel, then the first focusable element inside of that should get focus). For going to another Tab, the recommendation seems to suggest to use arrow keys and not the `TAB` key.

How should this work when the [Focusability of disabled controls](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols) suggests to use `aria-disabled` for Tabs and make sure that the tabs are reachable with keyboard through tabbing? I do not quite understand how the `aria-disabled` usage instead of `disabled` should behave when implemented with the current examples and feel like the recommendation contradicts the pattern example considering that "tab key press when on a Tab should go to the next element within the tab panel or focus the tab panel itself".

Can anyone give some clarification on how the aria-disabled on Tabs should be handled when following the APG practices since the APG example seems to not include examples of how this should be done.

Is it specifically that users should enable tabbing to the first tab even if it's disabled and then they are expected to tab an extra time to end up on the first non disabled tab? Or should they start using arrow keys once they end up on a disabled tab in this case? (Will it not be confusing to have users be communicated that they are on a disabled tab and also expect them to understand that they can use arrow keys to navigate to non disabled tabs in that case?)

**Comment by mcking65** [comment]
@jchiem asks:

> The APG pattern suggests that tabs should be reachable via the tab key, and then navigated using arrow keys. If a tab is disabled using aria-disabled, how should that behave in terms of keyboard navigation?

If a tab is disabled, you would make it focusable as described in the keyboard section of the tabs design pattern. Some important points to note:
1. Since a disabled tab is never the active tab, it would never be in the page tab sequence. Tabbing into a tablist puts focus on the currently active tab.
2. If tabs are activated automatically, focusing the disabled tab would not cause activation, so the tab that was previously focused will still be selected and its tab panel would continue to be displayed.
3. If tabs are activated manually, focusing the disabled tab and pressing Enter or Space would do nothing.

>Isn't it conflicting to make tabs focusable but have the tab pattern show tabs as not keyboard navigatable with the "tab key"?

There are no apparent conflicts because when moving focus into a tab list with the "Tab" key, focus always lands on the active tab, which should have aria-selected set to true.

> The current example shows that tabs should focus the first focusable tab within the tabs if you use `TAB` on your keyboard to navigate

That is not the behavior. The behavior is to focus the currently active tab. If you activate the second, third, or fourth tab in the example and then "Tab" out of the tab list in either direction, "Tabbing" back into the tab list will focus the active tab, not the first tab.

If the first tab were disabled, it could not be active, so it would not receive focus when tabbing with the tab key.

Does this answer your question?
